content/docs/concepts/resources/functions.md

```diff
@@ -131,34 +131,170 @@ variables:
+    // not happen until `res1` and `res2` have been processed. This will work for
+    // any set of resources, even those with no explicit outputs, since the `.Urn`
+    // output is always available.
+    Output.Tuple(res1.Urn, res2.Urn).Apply(t => myFunctionOutput());
+});
+```
+
+{{% /choosable %}}
+{{% choosable language go %}}
+
+```go
+package main
+
+import (
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		res1, err := NewMyResource(ctx, "res1", nil)
+		if err != nil {
+			return err
+		}
+
+		res2, err := NewMyResource(ctx, "res2", nil)
+		if err != nil {
+			return err
+		}
+
+		// Assuming `myFunctionOutput` is an output-form invocation of the `myFunction`
+		// provider function, this use of `All` and `ApplyT` will ensure that it does not
+		// happen until `res1` and `res2` have been processed. This will work for any set
+		// of resources, even those with no explicit outputs, since the `.URN` output is
+		// always available.
+		_, err = pulumi.All(res1.URN, res2.URN).ApplyT(func(args []interface{}) (interface{}, error) {
+			return myFunctionOutput()
+		})
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+}
+```
+
+{{% /choosable %}}
+{{% choosable language python %}}
+
+```python
+import pulumi
+
+res1 = MyResource("res1", {})
+res2 = MyResource("res2", {})
+
+# Assuming `my_function_output` is an output-form invocation of the `my_function`
+# provider function, this use of `all` and `apply` will ensure that it does not
+# happen until `res1` and `res2` have been processed. This will work for any set
+# of resources, even those with no explicit outputs, since the `.urn` output is
+# always available.
+pulumi.all(res1.urn, res2.urn).apply(lambda args: my_function_output())
+```
+
+{{% /choosable %}}
+{{% choosable language java %}}
+
+```java
+package generated_program;
+
+import com.pulumi.Context;
+import com.pulumi.Output;
+import com.pulumi.Pulumi;
+
+public class App {
+    public static void main(String[] args) {
+        Pulumi.run(App::stack);
+    }
+
+    public static void stack(Context ctx) {
+        final var res1 = new MyResource("res1", new MyResourceArgs());
+        final var res2 = new MyResource("res2", new MyResourceArgs());
+
+        // Assuming `myFunctionOutput` is an output-form invocation of the `myFunction`
+        // provider function, this use of `tuple` and `applyValue` will ensure that it does
+        // not happen until `res1` and `res2` have been processed. This will work for
+        // any set of resources, even those with no explicit outputs, since the `.urn`
+        // output is always available.
+        Output.tuple(res1.getUrn(), res2.getUrn()).applyValue(t -> myFunctionOutput());
+    }
+}
+```
+
+{{% /choosable %}}
+{{% choosable language yaml %}}
+
+Output form invocations are not yet supported in YAML
+
+{{% /choosable %}}
+{{< /chooser >}}
 
 ### Provider methods
```

Comment:
Smallish thing, but would it be clearer to call these Resource methods instead (since they're exposed on resources)? Seems so, but will leave the call up to you.

```suggestion
### Resource methods
```